dingo/bindings/hmc_sampling.h

```diff
@@ -0,0 +1,132 @@
+#include "bindings.h" 
+#include "ode_solvers/ode_solvers.hpp"
+#include <iostream> 
+#include <math.h> 
+#include <stdexcept> 
+using namespace std;
+
+
+// Initialization of the HPolytopeCPP class
+HPolytopeCPP::HPolytopeCPP() {}
+HPolytopeCPP::HPolytopeCPP(double *A_np, double *b_np, int n_hyperplanes, int n_variables){
+
+MT A;
+VT b;
+A.resize(n_hyperplanes,n_variables);
+b.resize(n_hyperplanes);
+
+int index = 0;
+for (int i = 0; i < n_hyperplanes; i++){
+   b(i) = b_np[i];
+   for (int j=0; j < n_variables; j++){
+      A(i,j) = A_np[index];
+      index++;
+   }
+}
+HP = Hpolytope(n_variables, A, b);                                 
+RNGType rng(HP.dimension());
+HP.normalize();
+int d = HP.dimension();
+   
+
+void hmc_leapfrog_gaussian(int walk_len,
+                                    int number_of_points, 
+                                    int number_of_points_to_burn, 
+                                    double* samples,
+                                    double variance,
+                                    double* bias_vector_,
+                                    double* inner_point) {
+                                                                        
+   Point starting_point; 
+   VT inner_vec(d);
+
+   for (int i = 0; i < d; i++){
+      inner_vec(i) = inner_point[i];
+   }
+   
+   Point inner_point2(inner_vec); 
+   CheBall = std::pair<Point, NT>(inner_point2, radius);
+   HP.set_InnerBall(CheBall);
```

Comment:
For example, the polytope is already initialized in `apply_sampling()`,you could pass this as input to this function, so you don't need the input `double* inner_point`